ext/ffi/lib.rs

```diff
@@ -2063,47 +2063,35 @@ fn op_ffi_call_nonblocking<'scope>(
   })
 }
 
-#[op(v8)]
-fn op_ffi_ptr_of<FP, 'scope>(
-  scope: &mut v8::HandleScope<'scope>,
+#[op(fast)]
+fn op_ffi_ptr_of<FP>(
   state: &mut deno_core::OpState,
-  buf: serde_v8::Value<'scope>,
-) -> Result<serde_v8::Value<'scope>, AnyError>
+  buf: &[u8],
+  out: &mut [u32],
+) -> Result<(), AnyError>
 where
   FP: FfiPermissions + 'static,
 {
   check_unstable(state, "Deno.UnsafePointer#of");
   let permissions = state.borrow_mut::<FP>();
   permissions.check(None)?;
 
-  let pointer = if let Ok(value) =
-    v8::Local::<v8::ArrayBufferView>::try_from(buf.v8_value)
-  {
-    let backing_store = value
-      .buffer(scope)
-      .ok_or_else(|| {
-        type_error("Invalid FFI ArrayBufferView, expected data in the buffer")
-      })?
-      .get_backing_store();
-    let byte_offset = value.byte_offset();
-    &backing_store[byte_offset..] as *const _ as *const u8
-  } else if let Ok(value) = v8::Local::<v8::ArrayBuffer>::try_from(buf.v8_value)
+  let outptr = out.as_ptr();
+  let outptr_integer = outptr as usize;
+  let length = out.len();
+  if length < (std::mem::size_of::<usize>() / std::mem::size_of::<u32>())
+    || (outptr_integer % std::mem::size_of::<usize>()) != 0
```

Comment:
```rust
#[op(fast)]
fn op_ffi_ptr_of<FP>(
  state: &mut deno_core::OpState,
  buf: &mut [u8],
  out: &mut [u32],
) -> Result<(), AnyError>
where
  FP: FfiPermissions + 'static,
{
  check_unstable(state, "Deno.UnsafePointer#of");
  let permissions = state.borrow_mut::<FP>();
  permissions.check(None)?;

  let ptr = buf.as_ptr() as u64;

  out[0] = ptr as u32;
  out[1] = (ptr >> 32) as u32;
  Ok(())
}
```
out[0]/out[1] will panic out of bounds so the length check is also not needed.

Reply:
Is it preferable that a panic occurs or should it rather be a throw?

Reply:
Also, I would presume that it'd be faster to write in a single "operation" rather than do the shifting etc. I was actually thinking of perhaps making the checks `debug_asserts`.

Reply:
yes panic is better when its not possible to invoke this from a public API. It forces users to report bugs

Reply:
I changed these to asserts and went with a write-to-pointer-of-u64 handling instead.